test/run_tests.sh

```diff
@@ -3,16 +3,27 @@
 set -o errexit
 set -o nounset
 set -o pipefail
+set -o xtrace
 
 # "TARGET" is a Makefile target that runs tests
 TARGET=$1
 
 INTEGRATION_OUTPUT_JUNIT=${INTEGRATION_OUTPUT_JUNIT:-false}
 
+# Set test harness artifacts dir to '/tmp/kudo-e2e-test', as it's easier to copy out from a container.
+echo 'artifactsDir: /tmp/kudo-e2e-test' >> kudo-e2e-test.yaml.tmpl
+
 # Pull the builder image with retries if it doesn't already exist.
 retries=0
 builder_image=$(awk '/FROM/ {print $2}' test/Dockerfile)
 
+function archive_logs() {
```

Comment:
Please move this function definition a little up or down. Here it's right in the middle of the docker pull retry logic.